### PR TITLE
Optimize memory usage in `TFTNativeModel` by eliminating copying during making samples

### LINKED ISSUE #493: Optimize memory usage in `TFTNativeModel` by eliminating copying during making samples
### 🚀 Task description

Now `TFTNativeModel` uses enormous amount memory when applied to big datasets. It occurs when samples are generating. It happens because samples consist of arrays copies instead of views, see https://github.com/etna-team/etna/blob/950f676c24e3f3b67917991f63200fa3482b9d84/etna/models/nn/tft_native/tft.py#L416

`.float` was implemented because all dataframe values were casted here https://github.com/etna-team/etna/blob/950f676c24e3f3b67917991f63200fa3482b9d84/etna/models/nn/tft_native/tft.py#L385 to one `object` dtype due to at least one existing `object` feature (ex. timestamp).

### Plan

To fix this, remove all unnecessary columns.
For example,
```
categ_columns = list(set(self.static_categoricals + self.time_varying_categoricals_encoder + self.time_varying_categoricals_decoder))
reals_columns = list(set(self.static_reals + self.time_varying_reals_encoder + self.time_varying_reals_decoder))
df = df[categ_columns + reals_columns]
```

### Test cases

Add one check to `test_tft_make_samples` whether views or copies are used
```
assert feature.base is not None
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by egoriyaa**
## Before submitting (must do checklist)

- [ ] Did you read the [contribution guide](https://github.com/etna-team/etna/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs? We use Numpy format for all the methods and classes.
- [ ] Did you write any new necessary tests?
- [ ] Did you update the [CHANGELOG](https://github.com/etna-team/etna/blob/master/CHANGELOG.md)?

## Closing issues
closes #493

### CODE REVIEW

File: etna/models/nn/tft_native/tft.py [hunk 1]
```diff
@@ -381,6 +381,17 @@ def make_samples(self, df: pd.DataFrame, encoder_length: int, decoder_length: in
...
+            set(
+                self.static_categoricals
+                + self.time_varying_categoricals_encoder
+                + self.time_varying_categoricals_decoder
+            )
+        )
+
+        df = df[reals_columns + categ_columns]
         column_to_index = {column: index for index, column in enumerate(df.columns)}
         values = df.values.T
```
Comment (d-a-bunin) [inline etna/models/nn/tft_native/tft.py:396]: As I understand correctly, the core problem is here when we more from pandas to numpy.

File: CHANGELOG.md [hunk 2]
```diff
@@ -39,7 +39,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 - Fix working with `embedding_sizes` in `202-NN_examples` notebook ([#489](https://github.com/etna-team/etna/pull/489))
 - Disallow dropping target in `TSDataset.drop_features` ([#491](https://github.com/etna-team/etna/pull/491))
-- 
+- Optimize memory usage in `TFTNativeModel` ([#494](https://github.com/etna-team/etna/pull/494))
```
Comment (d-a-bunin) [inline CHANGELOG.md:42]: We could write it more specific:
```
Optimize memory usage in `TFTNativeModel` by eliminating copying during making samples
```

We could also update issue name and PR name in a similar way.

File: tests/test_models/test_nn/tft_native/test_tft_native.py [hunk 3]
```diff
@@ -188,31 +188,37 @@ def test_tft_make_samples(
             df[[feature]].iloc[:1],
             first_sample["static_reals"][feature],
         )
+        assert first_sample["static_reals"][feature].base is not None
```
Comment (d-a-bunin) [inline tests/test_models/test_nn/tft_native/test_tft_native.py:191]: Don't we want to add such tests for other models too? It seems like a good practice to follow for future models.

**Review by d-a-bunin — CHANGES_REQUESTED** [review]

File: etna/models/nn/tft_native/tft.py (hunk 1 above)
Comment (egoriyaa) [reply etna/models/nn/tft_native/tft.py:396]: Yes

File: tests/test_models/test_nn/tft_native/test_tft_native.py (hunk 3 above)
Comment (egoriyaa) [reply tests/test_models/test_nn/tft_native/test_tft_native.py:191]: Good idea!

**Pull request merged**